src/parser.ts

```diff
@@ -138,6 +158,34 @@ class Parser {
             const componentName = computeComponentName(exp, source);
             const defaultProps = this.extractDefaultPropsFromComponent(exp, source);
             const props = this.getPropsInfo(propsType, defaultProps);
+            const { propFilter } = this.opts;
```

Comment:
What do you think about changing filtering array here with filter function that would be "build" in constructor.
Here it could be something:

```ts
const props = this
     .getePropsInfo(propsType, defaultProps) // now this would need to return PropInfo[]
     .filter(this.propsFilter); 

// later expanded back to object
const propObj = props.reduce((acc, prop) => {
     acc[prop.name] = prop;
     return acc;
}, {});
```
In constructor we could have something like 
```ts
this.propsFilter = buildFilter(opts: ParserOptions);
```
In that case we could easily write unit test only for `buildFilter` method which would be easier. Right now our tests are more kind of integration tests which is good for making sure the parser itself is working as expected but harder to cover all different settings for options for example. Then one integration test for filter would be enough and rest can be covered by unit tests.

Reply:
forgot to pass component to `propsFilter` but you get the idea

Reply:
Good idea. I've updated the PR to extract the `buildFilter` method. All filters are covered by unit tests. Technically the integration tests are not necessary anymore, so feel free to delete those that aren't necessary anymore in your opinion.